### PR TITLE
Add program for goal-oriented Maxwell eigenvalue hp-refinement

### PR DESCRIPTION
**Pull request by harmonj**
Adds a code gallery project for goal-oriented hp-refinement for eigenvalue quantities of interest of the Maxwell eigenvalue problem for a 2-D benchmark that includes eigenpairs with singular eigenfunctions.

This is tested and working with the latest version of deal.II (9.4).

### CODE REVIEW

**Comment by bangerth** [comment]
@harmonj I haven't found the time to look at your program yet :-( But it hasn't slipped my mind and I look forward to getting to it in the next two or three weeks!

**Comment by bangerth** [comment]
I am embarrassed that it took me another six months to come back to this when I said "two or three weeks". I will look through this now, and if it looks reasonable will merge and fix small things myself.

File: Maxwell-Eigenvalue-hp-Refinement/maxwell-hp.cc [hunk 1]
```diff
@@ -0,0 +1,2271 @@
...
+  eigenfunctions
+  */
+  template <int dim>
+  unsigned int
+  EigenSolver<dim>::solve_problem()
+  {
+    setup_system();
+    assemble_system();
+
+    SolverControl                    solver_control(dof_handler.n_dofs() * 10,
```
Comment (bangerth) [inline Maxwell-Eigenvalue-hp-Refinement/maxwell-hp.cc:454]:
```suggestion
    SolverControl solver_control(dof_handler.n_dofs() * 10,
```

**Review by bangerth — APPROVED** [review]
I've looked through the code and this is good enough. It follows step-14 reasonably closely.

Nicely done!

**Comment by bangerth** [comment]
I made a couple of minor changes to the readme as well, but only mark-up stuff.

**Comment by bangerth** [comment]
I can confirm that the program compiles and runs...at least somewhat. I'll have a small fix to avoid a warning in a minute (see #119).

@harmonj In debug mode, the program is stuck at the stage where it says
```
Executing refinement for the Kelly strategy!
Initializing RegularityIndicator...
(This may take a while if the max expansion order is set too high)
```
for over an hour. Can we change the defaults so that in debug mode, this step finishes in an acceptable time? Or at least keeps printing status updates so we know that the program continues to do something useful?

**Comment by harmonj** [comment]
@bangerth Thank you for getting this added! The defaults should be reduced significantly, otherwise it will take a long time in debug mode. Lines 2224, 2227 the max expansion order (currently set to 12) should be reduced to around 5. I will take a closer look tonight and make any other changes needed.

**Comment by bangerth** [comment]
Great! I would gladly accept a pull request (and do that on shorter notice :-) or, if you want me to, can also do it myself.

For the record, in release mode the program finishes in under 9 minutes.

**Pull request merged**